Change title for report type on "My data" -> List

DigitalSeneca opened the pull request:
fixes #63 

| English | Spanish |
| ------------- | ------------- |
| ![image](https://github.com/Mosquito-Alert/Mosquito-Alert-Mobile-App/assets/157690872/98d6fe8c-d35f-46ea-a386-efa2ac953b05) | ![image](https://github.com/Mosquito-Alert/Mosquito-Alert-Mobile-App/assets/157690872/e88ec79c-461f-49a7-a1ae-53305f2acee3) |
| ![image](https://github.com/Mosquito-Alert/Mosquito-Alert-Mobile-App/assets/157690872/ab760176-d343-432b-b533-97d5a80202f8) | - |

epou commented on lib/pages/my_reports_pages/components/reports_list_widget.dart:
```diff
@@ -12,6 +12,21 @@ class ReportsList extends StatelessWidget {
+      translationString = 'single_mosquito';
+    }else if(reportType == 'bite'){
+      translationString = 'single_bite';
+    }else if(reportType == 'bite'){
```
'bite' already considered. Change to "breeding_site"

epou commented on lib/pages/my_reports_pages/components/reports_list_widget.dart:
```diff
@@ -12,6 +12,21 @@ class ReportsList extends StatelessWidget {
 
+  String getTranslatedReportType(BuildContext context, String? reportType){
+    var translationString = '';
+    if(reportType == 'adult'){
+      translationString = 'single_mosquito';
+    }else if(reportType == 'bite'){
+      translationString = 'single_bite';
+    }else if(reportType == 'bite'){
+      translationString = 'single_breeding_site';
+    }else{
+      print('Unhandled report type: $reportType');
+    }
```
Maybe using a switch - case would improve readability....

```dart
switch (reportType) {
    case 'adult':
      translationString = 'single_mosquito';
      break;
    case 'bite':
      translationString = 'single_bite';
      break;
    case 'breeding_site':
      translationString = 'single_breeding_site';
      break;
    default:
      print('Unhandled report type: $reportType');
      return reportType ?? '';
  }
```

epou commented on lib/pages/my_reports_pages/components/reports_list_widget.dart:
```diff
@@ -12,6 +12,21 @@ class ReportsList extends StatelessWidget {
   ReportsList(this.reports, this.onTap);
 
+  String getTranslatedReportType(BuildContext context, String? reportType){
+    var translationString = '';
```
`String translationString;`

epou commented on lib/pages/my_reports_pages/components/reports_list_widget.dart:
```diff
@@ -12,6 +12,21 @@ class ReportsList extends StatelessWidget {
 
   ReportsList(this.reports, this.onTap);
 
+  String getTranslatedReportType(BuildContext context, String? reportType){
```
Does `reportType` can be null in the code?

DigitalSeneca replied:
Right now it can theoretically be null, as defined in the Reports class. That's because after flutter 3.0.0 all types require null safety.

Maybe it's better to change the nomenclature to this so that it can't be null, but if we decide to do this, then we have to handle it correctly for this variable and also the others which are nullable right now
```
class Report {
  // other vars
  late String type;  // before: String? type;

  Report(
      // other vars
      required this.type, // before: this.type,
```

epou replied:
In the backend `Report.type` is a mandatory field (see: https://github.com/Mosquito-Alert/mosquito_alert/blob/e0618655f6320c9befb29b0741a35fa096fe8c64/tigaserver_app/models.py#L416) so it would make sense to make it required in the app code too. If you think we might address this in a new issue we can create a one for checking the required variables in the code.

DigitalSeneca replied:
> If you think we might address this in a new issue we can create a one for checking the required variables in the code.

Yeah I think it makes sense to address all or most in a dedicated pull request

epou commented on lib/pages/my_reports_pages/components/reports_list_widget.dart:
```diff
@@ -57,22 +72,24 @@ class ReportsList extends StatelessWidget {
-                                  color: Colors.grey),
+                                MyLocalizations.of(context, 'at_time_txt')! +
+                                ' ' +
+                                DateFormat('dd-MM-yyyy')
```
Use `dd-MM-yyyy HH:mm`

epou commented on lib/pages/my_reports_pages/components/reports_list_widget.dart:
```diff
@@ -57,22 +72,24 @@ class ReportsList extends StatelessWidget {
+                                  .format(DateTime.parse(
+                                      reports[index].creation_time!))
+                                  .toString() +
+                                ' ' +
+                                DateFormat.Hm()
+                                  .format(DateTime.parse(
+                                      reports[index].creation_time!))
+                                  .toString(),
```
Remove this after adding `dd-MM-yyyy HH:mm` to the previous `DateFormat`

epou reviewed (changes requested):
Some improvements are suggested.

Not sure if when clicking a report on the list, the title of the 'detailed report view' is also changed to {Report.type} or not :thinking:

DigitalSeneca commented:
@epou As the screenshot shows, right now the type only works for English and is empty for other languages. This will be fixed just by adding the strings with #69 

"reports[index].displayCity" seems to be null in some cases. This was already happening before. How should this case be handled?

epou commented:
Could you implement a fallback to the English version when no translation is available in the specified language? I prefer seeing an English word instead of an empty value. 

Regarding `displayCity`, there's an issue reported (refer to issue #77).

DigitalSeneca commented:
> Some improvements are suggested.
> 
> Not sure if when clicking a report on the list, the title of the 'detailed report view' is also changed to {Report.type} or not 🤔

I have updated it, when clicking a report, it also shows {Report.type} as title instead of "Notification of the day"

> Regarding displayCity, there's an issue reported (refer to issue https://github.com/Mosquito-Alert/Mosquito-Alert-Mobile-App/issues/77).

Okey, I didn't see it. I'll handle it in a separate pull request

epou commented on assets/language/en.json:
```diff
@@ -164,7 +164,7 @@
     "list_txt": "List",
     "report_of_the_day_txt": "Notification of the day ",
-    "location_txt": "Approximate location",
+    "location_txt": "Approximate location:",
```
Remove `:`

DigitalSeneca replied:
okey, I'll trust the provided translations. I was just confused that some languages have the : on this string while others don't

epou replied:
You're right, some translations do contain ":". I will solve that on issue #80 ensuring that these characters are on the code side, not on the translation. Keep it without the ":" for this pull request,  thought

DigitalSeneca replied:
perfect

epou commented on lib/pages/my_reports_pages/components/reports_list_widget.dart:
```diff
@@ -57,22 +58,19 @@ class ReportsList extends StatelessWidget {
+                                  Utils.getTranslatedReportType(context, reports[index].type),
                                   fontSize: 14),
                               Style.body(
                                   '${MyLocalizations.of(context, "location_txt")} ${reports[index].displayCity ?? ''}'),
```
Add `:` after the location_txt translated string

DigitalSeneca replied:
should I remove the `:` from the strings where it's included, like ca, es, eu, gl, pt? I am afraid there would be a double `:` otherwise

epou replied:
I've already removed them from the localise.biz platform. So in the next export, it will be already removed :)

epou reviewed (changes requested):
Just change where the `:` is

epou commented:
Ei! I've already given my approval, but it's still a Draft...
